src/NuGet.Core/NuGet.LibraryModel/LibraryExtensions.cs

```diff
@@ -16,9 +16,9 @@ public static bool IsEclipsedBy(this LibraryRange library, LibraryRange other)
                 && string.Equals(library.Name, other.Name, StringComparison.OrdinalIgnoreCase);
         }
 
-        public static T GetItem<T>(this Library library, string key)
+        public static T? GetItem<T>(this Library library, string key)
         {
-            object value;
+            object? value;
```

Comment:
I like this syntax better personally if you want to update this class to use it:

```c#
return library.Items.TryGetValue(key, out object? value) ? value as T : default(T);
```

Reply:
Firstly, `value as T` is a pet peeve of mine. 😁 Also see [this](https://github.com/NuGet/NuGet.Client/pull/5309). Unfortunately, this method allows nulls to be returned (otherwise `default(T)` would result in a compiler error, now that this project has nullable enabled), so `value as T` won't trigger the nullable compiler error. However, if the value has the wrong type, it will silently be returned as null, and the app will have the wrong behaviour, which will be difficult to diagnose. Using `(T)value` instead will throw an InvalidCastException, which will make it very clear what went wrong, and the line of code and stack trace, making it very easy to diagnose.

Secondly, as mentioned in another comment a few seconds ago, in order to save time (other projects took many hours to fix compiler errors), I didn't spend time refactoring.

I can make these small changes if preferred, but that's the reason I didn't before creating the PR.